search/search_service/proxy/es_proxy_v3.py

```diff
@@ -0,0 +1,310 @@
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 2
+                }),
+                Match(chart_names={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 2
+                }),
+                Match(uri={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 4
+                }),
+            ])
+        elif resource == Resource.FEATURE:
+            should_clauses.extend([
+                Match(feature_group={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 3
+                }),
+                Match(version={
+                    "query": query_term
+                }),
+                Match(entity={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 2
+                }),
+                Match(status={
+                    "query": query_term
+                }),
+            ])
+        elif resource == Resource.USER:
+            # replaces rather than extending
+            should_clauses = [
+                Match(name={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 5
+                }),
+                Match(first_name={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 3
+                }),
+                Match(last_name={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 3
+                }),
+                Match(team_name={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10
+                }),
+                Match(key={
+                    "query": query_term,
+                    "fuzziness": DEFAULT_FUZZINESS,
+                    "max_expansions": 10,
+                    "boost": 4
+                }),
+            ]
+
+        must_clauses: List[Q] = [Q(BOOL_QUERY, should=should_clauses)]
+
+        return must_clauses
+
+    def _build_should_query(self, resource: Resource, query_term: str) -> List[Q]:
+
+        # no scoring happens if there is no search term
+        if query_term == '':
+            return []
+
+        # general usage metric for searcheable resources
+        usage_metric_fields = {
```

Comment:
nit: rename to usage_metric_boosts. Otherwise makes it look like you're hard-coding usage metrics.